src/music.h

```diff
@@ -0,0 +1,23 @@
+#include "raylib-cpp.hpp"
+#include <memory>
+#include <nlohmann/json.hpp>
+#include <string>
+
+const std::string MUSIC_CONFIG_PATH =  "../assets/music/";
+
+class Sounds {
+  nlohmann::json config;
+  //raylib::AudioDevice audio;
+
+  //std::unique_ptr<raylib::Sound> walk;
+  std::unique_ptr<raylib::Music> menu; // loops by default
```

Comment:
why do these need to be `unique_ptr`?

Reply:
I've been stuck with this  --- what's the idiomatic way to instantiate classes inside another class's constructor
or rather,

I want to do something like 

```c++
class ASound {
  raylib::Sound walk;
  
public:
    ASound();
    ~ASound();   
}

ASound::ASound(){
   nlohmann::json config;
...
   // load json config
...
   walk = raylib::Sound(config["walk"]);
}

ASound::play_walk(){
  walk.Play();
}
```

but compiler goes brrrrrrrrrrrrrrrrrr

Reply:
why not have an `unordered_map` of sounds? we don't know what sounds we will want, so why hardcode them into your object?

Reply:
wait we could do something like 
```c++
class ASound {
  raylib::Sound walk ("path/to/sound.wav");
  
```

right?

Reply:
you might be able to do:
```cpp
class ASound {
    raylib::Sound walk { "path/to/sound.wav" }
```
or

```cpp
    ASound::ASound(): walk("path/to/sound.wav") {

   }
```